### ISSUE TITLE
Investigate ls -Q option to quote file names

### ISSUE THREAD

**Issue by csams** [RedHatInsights/insights-core#1530]
Some really rare file names include control characters and even newlines. So far we've only seen one instance of names with [newlines embedded at the end](#1527), which we've [accounted for](#1529), but weird characters elsewhere could still pose a problem.

@PaulWay suggested `ls -Q` to quote the names, and I think this is worth investigating if we notice further issues.

**Comment by ritwik12** [comment]
@csams What are we thinking here as a solution? I got the problem and I think we can clean file names which consist of  `control characters, newlines, etc`. Let me know if this will work.

**Comment by PaulWay** [comment]
I'd also suggest writing a new style of data collector that used `opendir()` and `readdir()` calls to read a given directory directly.  This would avoid all the problems of parsing the output of `ls`.  It would then output data in a clearly defined, e.g. as JSON, which could then be read by a suitable parser.

**Comment by csams** [comment]
@PaulWay yep, that's the intent. `ls` is notoriously difficult to parse. Once we get core3 client collection going, rewriting the `ls` code will be on the list of things to tackle.
